### PR TITLE
Adds filter functionality for the new data aggregation pages

### PR DESCRIPTION
**Pull request by andrea-kyurchiev**
### What

Adds the filters and validation for the new data aggregation pages

### How to review

Requires: 
[https://github.com/andrea-kyurchiev/dp-frontend-search-controller/tree/feat/reworks-data-aggregation-pages](https://github.com/andrea-kyurchiev/dp-frontend-search-controller/tree/feat/reworks-data-aggregation-pages)
[https://github.com/andrea-kyurchiev/dp-search-api/tree/feat/apply-date-filters-to-query](https://github.com/andrea-kyurchiev/dp-search-api/tree/feat/apply-date-filters-to-query)
[https://github.com/andrea-kyurchiev/dp-frontend-router/tree/feat/adds-data-aggregation-routes](https://github.com/andrea-kyurchiev/dp-frontend-router/tree/feat/adds-data-aggregation-routes)

### Who can review

Anyone

### CODE REVIEW

File: src/js/dp/timeseries.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        exitBtn = timeSeriesContainer.querySelector(".timeseries__list--exit"),
+        rememberCb = timeSeriesContainer.querySelector("#remember-selection");
+        
+    exitBtn.addEventListener("click", () => {
+        boxWithStuff.classList.toggle("hidden");
+    });
+
+    function initialize() {
+        remember = getLocalStorageProperty(rememberCookieName);
+        if (remember) {
```
Comment (jondewijones) [inline src/js/dp/timeseries.js:26]: Do we want/need two `if (remember)` straight after each other (line 26 & 33)? Any reason the logic couldn't be in the same block?

File: src/js/dp/timeseries.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            return; // it is already in the list    
+        }
+        
+        timeseriesList[timeseries.uri] = timeseries;
+        addToPage(timeseries)
+    }
+    
+    function removeElement(uri) {
+        listCount--;
+        // var timeseries = timeseriesList[uri];
```
Comment (jondewijones) [inline src/js/dp/timeseries.js:142]: If not needed can we remove?

Comment (andrea-kyurchiev) [reply src/js/dp/timeseries.js]: Removed

File: src/js/dp/timeseries.js (hunk 1 above)
Comment (andrea-kyurchiev) [reply src/js/dp/timeseries.js]: Apologies, it was supposed to check if remember is defined on line 26. Fixed now!

**Review by jondewijones — COMMENTED** [review]
Could we use `const` or `let` to make it consistent with other files in the design system?

**Review by jondewijones — APPROVED** [review]

**Comment by lindenmckenzie** [comment]
Cherrypicked across to https://github.com/ONSdigital/dp-design-system/pull/185

**Pull request closed without merging**